akka-cluster/src/main/scala/akka/cluster/Gossip.scala

```diff
@@ -165,35 +166,46 @@ private[cluster] final case class Gossip(
    *
    * @return true if convergence have been reached and false if not
    */
-  def convergence(selfUniqueAddress: UniqueAddress, exitingConfirmed: Set[UniqueAddress]): Boolean = {
+  def convergence(team: Team, selfUniqueAddress: UniqueAddress, exitingConfirmed: Set[UniqueAddress]): Boolean = {
     // First check that:
-    //   1. we don't have any members that are unreachable, excluding observations from members
+    //   1. we don't have any members that are unreachable in the team, excluding observations from members
     //      that have status DOWN, or
     //   2. all unreachable members in the set have status DOWN or EXITING
     // Else we can't continue to check for convergence
     // When that is done we check that all members with a convergence
     // status is in the seen table, i.e. has seen this version
-    val unreachable = reachabilityExcludingDownedObservers.allUnreachableOrTerminated.collect {
-      case node if (node != selfUniqueAddress && !exitingConfirmed(node)) ⇒ member(node)
-    }
-    unreachable.forall(m ⇒ Gossip.convergenceSkipUnreachableWithMemberStatus(m.status)) &&
-      !members.exists(m ⇒ Gossip.convergenceMemberStatus(m.status) &&
-        !(seenByNode(m.uniqueAddress) || exitingConfirmed(m.uniqueAddress)))
+    val unreachableInTeam = teamReachabilityExcludingDownedObservers(team).allUnreachableOrTerminated.collect {
+      case node if node != selfUniqueAddress && !exitingConfirmed(node) ⇒ member(node)
+    }.filter(_.team == team)
```

Comment:
I can't because it relies on the member(node) lookup which is done after matching.